compiler/rustc_privacy/src/lib.rs

```diff
@@ -1207,6 +1215,28 @@ impl<'tcx> Visitor<'tcx> for TypePrivacyVisitor<'tcx> {
         intravisit::walk_ty(self, hir_ty);
     }
 
+    fn visit_infer(&mut self, inf: &'tcx hir::InferArg) {
+        self.span = inf.span;
+        if let Some(typeck_results) = self.maybe_typeck_results {
+            if let Some(ty) = typeck_results.node_type_opt(inf.hir_id) {
+                if self.visit(ty).is_break() {
+                    return;
+                }
+            }
+        } else {
+            let local_id = self.tcx.hir().local_def_id(inf.hir_id);
+            if self.tcx.opt_const_param_of(local_id).is_some() {
```

Comment:
can you add a `FIXME(generic_arg_infer)` here so that this is easily grepable?

Reply:
Updated to visit the def id, not sure why I wasn't doing that before